src/package-scanner.ts

```diff
@@ -69,26 +74,84 @@ export class PackageScanner {
   }
 
   /**
-   * Scan a package.
+   * Scan a package and return the scan result. This method will first try to
+   * fetch a package manifest from npm. Failing that (no manifest exists, npm
+   * cannot be reached, etc.) it will scan the package manually.
    */
-  scanPackage(): PackageScanResult {
-    // Add this package to our cache so that it won't get double scanned.
-    // Gather all relevent package documents, and run the scanner on them.
-    for (const document of this.getPackageDocuments()) {
+  async scanPackage(): Promise<PackageScanResult> {
+    const resultsFromManifest = await this.getResultsFromManifest();
+    if (resultsFromManifest) {
+      this.scanPackageFromManifest(resultsFromManifest);
+    } else {
+      this.scanPackageManually();
+    }
+    return this.getResults();
+  }
+
+  /**
+   * Get a package manifest (a serializable version of the scanner results) for
+   * a package.
+   */
+  private scanPackageFromManifest(packageScanManifest: PackageScanResult) {
+    for (const [originalUrl, scanResult] of packageScanManifest.files) {
+      this.results.set(originalUrl, scanResult);
+      if (scanResult.type === 'js-module') {
+        for (const expr of scanResult.exportMigrationRecords) {
+          if (!this.namespacedExports.has(expr.oldNamespacedName)) {
+            this.namespacedExports.set(
+                expr.oldNamespacedName,
+                new JsExport(scanResult.convertedUrl, expr.es6ExportName));
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Scan each document in a package manually.
+   */
+  scanPackageManually() {
+    for (const document of this.getPackageHtmlDocuments()) {
       this.scanDocument(document);
     }
+  }
 
-    return this.getResults();
+  /**
+   * Fetch a conversion manifest from NPM. If none can be found, return null.
+   */
+  async getResultsFromManifest(): Promise<PackageScanResult|null> {
+    const npmPackageInfo = lookupDependencyMapping(this.packageName);
+    if (!npmPackageInfo) {
+      return null;
+    }
+
+    try {
+      const unpkgResponse = await fetch(`https://unpkg.com/${
+          npmPackageInfo.npm}@${npmPackageInfo.semver}/manifest.json`);
+      const manifestJson: PackageScanResultJson = await unpkgResponse.json();
+      const [allFiles, allExports] = filesJsonObjectToMap(
+          this.packageName, npmPackageInfo.npm, manifestJson, this.urlHandler);
+      return {
+        files: allFiles,
+        exports: allExports,
+      };
+    } catch (err) {
+      return null;
+    }
   }
 
   /**
    * Get all relevant HTML documents from a package that should be scanned,
    * coverted, or otherwise handled by the modulizer.
    */
-  getPackageDocuments() {
+  getPackageHtmlDocuments() {
     return [
-      ...this.analysis.getFeatures(
-          {kind: 'html-document', externalPackages: true})
+      ...this.analysis.getFeatures({
+        // Set externalPackages=true so that this method works on dependencies
+        // as well.
```

Comment:
this comment and the one below don't exactly say why we include external here and exclude it in the filter